youxia-q2seqware/src/main/java/io/cloudbindle/youxia/q2seqware/ExternalSeqwareScheduler.java

```diff
@@ -0,0 +1,136 @@
+package io.cloudbindle.youxia.q2seqware;
+
+import io.cloudbindle.youxia.util.Log;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+import org.apache.commons.cli.BasicParser;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.apache.commons.exec.CommandLine;
+import org.apache.commons.exec.DefaultExecutor;
+import org.apache.commons.exec.ExecuteException;
+import org.apache.commons.exec.PumpStreamHandler;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.support.GenericXmlApplicationContext;
+
+public class ExternalSeqwareScheduler {
+    private static final String SPRING_CONFIG_FILE = "q2seqware-spring-config.xml";
+    private static SeqwareStatusMonitor monitor = new SeqwareStatusMonitor();
+    private static String workflowName = null;
+    private static String workflowVersion = null;
+    
+    
+    public static void main(String[] args) {
+        ApplicationContext context = new GenericXmlApplicationContext(SPRING_CONFIG_FILE);
+        
+        getCLIArgs(args);
+
+        if (workflowName == null || workflowName.trim().equals("")) {
+            Log.error("Please specify a workflow name using the \"workflow\" option");
+        } else if (workflowVersion == null || workflowVersion.trim().equals("")) {
+            Log.error("Please specify a workflow version using the \"version\" option");
+        } else {
+            // 1) Query SeqWare to see if it can accept a new job.
+            String accessionID = monitor.getAccessionID(workflowName, workflowVersion);
+            // String status = monitor.checkSeqwareStatus(accessionID);
+            if (accessionID != null) {
+                // 2) If SeqWare isn't busy, get a message from the Queue/MessageSource and call SeqWare.
+                // if (status.equals("") || !status.equals("running"))
+                callSeqware(context, accessionID);
+            } else {
+                Log.error("Could not determine accession ID for workflow with name:" + workflowName + "; version: " + workflowVersion
+                        + ". Please check that the workflow you specified is installed correctly. Cannot proceed, program ending.");
+            }
+        }
+        ((ConfigurableApplicationContext) context).close();
+    }
+
+    private static void callSeqware(ApplicationContext context, String accessionID) {
+        if (canScheduleNewJobs(workflowName, workflowVersion)) {
+            // SeqwareJobMessageSource listener = new RabbitMQListener();
+            // SeqwareJobMessageSource listener = new SeqwareCGIMessageSource();
+            SeqwareJobMessageSource listener = context.getBean(SeqwareCGIMessageSource.class);
+            // listener.setSourceHost("localhost");
```

Comment:
Try not to check in commented code. If you have to, add in a comment with an explanation.
 
Is there a good reason for this to be here? Otherwise remove it, same with the big piece at the end in RabbitMQListener.